corehq/apps/repeaters/models.py

```diff
@@ -509,85 +552,39 @@ def try_now(self):
         return not self.succeeded
 
     def get_payload(self, save_failure=True):
-        try:
-            return self.repeater.get_payload(self)
-        except ResourceNotFound as e:
-            # this repeater is pointing at a missing document
-            # quarantine it and tell it to stop trying.
-            logging.exception(
-                u'Repeater {} in domain {} references a missing or deleted document!'.format(
-                    self._id, self.domain,
-                ))
-
-            if save_failure:
-                self._payload_exception(e, reraise=False)
-        except Exception as e:
-            if save_failure:
-                self._payload_exception(e, reraise=True)
-            else:
-                raise
+        warnings.warn("RepeatRecord.get_paylod is deprecated. Use Repeater._get_payload instead.")
+        return self.repeater.get_payload_and_handle_exception(self, save_failure=save_failure)
 
-    def _payload_exception(self, exception, reraise=False):
+    def handle_payload_exception(self, exception):
         self.succeeded = False
         self.failure_reason = unicode(exception)
         self.save()
-        if reraise:
-            raise
 
     def fire(self, force_send=False):
-        headers = self.repeater.get_headers(self)
-        auth = self.repeater.get_auth()
         if self.try_now() or force_send:
             self.overall_tries += 1
-            tries = 0
-            post_info = PostInfo(self.get_payload(), headers, auth)
-            self.post(post_info, tries=tries)
+            self.repeater.post_for_record(self)
```

Comment:
what if we want to define other methods for our repeater instead of POST?
e.g. SOAP or GET or some other thing? would you name this something different (e.g. `fire_for_record`)?